operator/cmd/run/run.go

```diff
@@ -302,6 +303,9 @@ func Command() *cobra.Command {
 	cmd.Flags().StringVar(&cloudSecretsGCPProjectID, "cloud-secrets-gcp-project-id", "", "GCP project ID in which the secrets are stored")
 	cmd.Flags().StringVar(&cloudSecretsAzureKeyVaultURI, "cloud-secrets-azure-key-vault-uri", "", "Azure Key Vault URI in which the secrets are stored")
 
+	// cluster restart on config change
+	cmd.Flags().BoolVar(&restartClusterOnConfigChange, "restart-cluster-on-config-change", false, "Set to true to restart the cluster when the config changes. This will add a pod annotation with a hash of the config. This is only used if the operator is running in operator V2 mode.")
```

Comment:
I think this would be better served as an annotation on the redpanda CR. My preference is to put feature toggles as close to the item the affect as possible. Though let me know if a CLI flag is easier to toggle in install packs.

Either way, please update the description to indicate that only cluster configs that require a restart will affect this hash or don't mention the implementation here at all.